carp.deployments.core/src/commonMain/kotlin/dk/cachet/carp/deployments/application/DeploymentService.kt

```diff
@@ -40,17 +40,23 @@ interface DeploymentService : ApplicationService<DeploymentService, DeploymentSe
 
 
     /**
-     * Instantiate a study deployment for a given [StudyProtocolSnapshot]
-     * and invite participants defined in [invitations].
+     * Instantiate a study deployment for a given [StudyProtocolSnapshot] with participants defined in [invitations].
+     *
+     * The identities specified in the invitations are used to invite and authenticate the participants.
+     * In case no account is associated to an identity, a new account is created for it.
+     * An invitation (and account details) is delivered to the person managing the identity,
+     * or should be handed out manually to the relevant participant by the person managing the identity.
      *
      * @throws IllegalArgumentException when:
      *  - [protocol] is invalid
-     *  - any of the master devices a participant in [invitations] is invited to is not part of [protocol]
+     *  - [invitations] is empty
+     *  - any of the assigned device roles in [invitations] is not part of the study [protocol]
+     *  - not all master devices part of the study [protocol] have been assigned a participant
      * @return The [StudyDeploymentStatus] of the newly created study deployment.
      */
     suspend fun createStudyDeployment(
         protocol: StudyProtocolSnapshot,
-        invitations: List<ParticipantInvitation> = emptyList()
+        invitations: List<ParticipantInvitation>
```

Comment:
Since participants are now instantly invited as part of the `createStudyDeployment` call, we should also pass optional `DeviceRegistration` for the assigned devices. Otherwise "device preregistration" is supported in a weird way. Preregistering a device would require creating a deployment, and then quickly adding a registration for that deployment, before the client has a chance of retrieving the master device deployment.

Logically, it makes more sense the _pre_registration is known already when the deployment is created.